### PR TITLE
add versioning policy to docs

### PR DESCRIPTION
**Pull request by sw-yx**
as agreed internally

### CODE REVIEW

File: docs/server-versions-dependencies.md [hunk 1]
```diff
@@ -10,6 +10,23 @@ This page details some of the version specific requirements and dependencies nee
 
 All versions of Temporal require that [Go v1.15+](https://golang.org/dl/) is installed in the environment.
 
+## Temporal Versioning and Support Policy
+
+All of Temporal's releases abide by [semver](https://semver.org/), but here we detail the precise upgrade paths and support we offer:
```
Comment (sw-yx) [inline docs/server-versions-dependencies.md:15]:
```suggestion
All of Temporal's releases abide by [semver](https://semver.org/), but here we detail the precise upgrade paths and support we offer, starting from Temporal `v1.7.0`:
```

File: docs/server-versions-dependencies.md [hunk 2]
```diff
@@ -10,6 +10,23 @@ This page details some of the version specific requirements and dependencies nee
 
 All versions of Temporal require that [Go v1.15+](https://golang.org/dl/) is installed in the environment.
 
+## Temporal Versioning and Support Policy
+
+All of Temporal's releases abide by [semver](https://semver.org/), but here we detail the precise upgrade paths and support we offer, starting from Temporal `v1.7.0`:
+
+- We define maintenance support as continuing to release critical bugfixes — related to security, preventing data loss, reliability — for previously published minor and major versions, whenever they are found.
```
Comment (flossypurse) [inline docs/server-versions-dependencies.md:17]:
```suggestion
- We provide maintenance support for previously published minor and major versions by continuing to release critical bug fixes related to security, the prevention of data loss, and reliability, whenever they are found.
```

File: docs/server-versions-dependencies.md [hunk 3]
```diff
@@ -10,6 +10,23 @@ This page details some of the version specific requirements and dependencies nee
...
+- **For Temporal Server OSS**
```
Comment (flossypurse) [inline docs/server-versions-dependencies.md:18]:
```suggestion
**Temporal Server OSS**
```

File: docs/server-versions-dependencies.md [hunk 4]
```diff
@@ -10,6 +10,23 @@ This page details some of the version specific requirements and dependencies nee
...
+    - We will publish incremental upgrade guides for each minor and major version
```
Comment (flossypurse) [inline docs/server-versions-dependencies.md:19]:
```suggestion
    - We aim to publish incremental upgrade guides for each minor and major version, which will include specifics around dependency upgrades that we have tested for (e.g. Cassandra 3.0 -> 3.11).
```

File: docs/server-versions-dependencies.md [hunk 5]
```diff
@@ -10,6 +10,23 @@ This page details some of the version specific requirements and dependencies nee
...
+        - This also includes specifying what dependency upgrades we test for (e.g. from Cassandra 3.0 to 3.11)
```
Comment (flossypurse) [inline docs/server-versions-dependencies.md:20]: As my recommendation above suggest, drop this bullet and just combine with the sentence above.

File: docs/server-versions-dependencies.md [hunk 6]
```diff
@@ -10,6 +10,23 @@ This page details some of the version specific requirements and dependencies nee
...
+    - We will offer maintenance support of the last 3 **minor** versions post release
```
Comment (flossypurse) [inline docs/server-versions-dependencies.md:21]:
```suggestion
    - We will offer maintenance support of the last 3 **minor** versions after a release, and do not plan to "backport" patches beyond that.
```

File: docs/server-versions-dependencies.md [hunk 7]
```diff
@@ -10,6 +10,23 @@ This page details some of the version specific requirements and dependencies nee
...
+        - In other words, we will not backport patches beyond the previous 3 minor versions for free
```
Comment (flossypurse) [inline docs/server-versions-dependencies.md:22]: As my recommendation above suggests, drop this bullet and just combine with the sentence above.

File: docs/server-versions-dependencies.md [hunk 8]
```diff
@@ -10,6 +10,23 @@ This page details some of the version specific requirements and dependencies nee
...
+    - We will offer maintenance support of **major** versions for at least 12 months after GA release, and provide at least 6 months' notice before EOL/deprecating support
```
Comment (flossypurse) [inline docs/server-versions-dependencies.md:23]:
```suggestion
    - We will offer maintenance support of **major** versions for at least 12 months after a GA release, and provide at least 6 months' notice before EOL/deprecating support.
```

File: docs/server-versions-dependencies.md [hunk 9]
```diff
@@ -10,6 +10,23 @@ This page details some of the version specific requirements and dependencies nee
...
+- **For Temporal Cloud**
```
Comment (flossypurse) [inline docs/server-versions-dependencies.md:24]:
```suggestion
**Temporal Cloud**
```

File: docs/server-versions-dependencies.md [hunk 10]
```diff
@@ -10,6 +10,23 @@ This page details some of the version specific requirements and dependencies nee
...
+    - We will automatically upgrade all customers to the latest minor versions
```
Comment (flossypurse) [inline docs/server-versions-dependencies.md:25]:
```suggestion
    - All customers will automatically be upgrade to the latest minor version.
```

File: docs/server-versions-dependencies.md [hunk 11]
```diff
@@ -10,6 +10,23 @@ This page details some of the version specific requirements and dependencies nee
...
+    - We will offer maintenance support of major versions for at least 18 months after GA release, and provide at least 6 months' notice before EOL/deprecating support
```
Comment (flossypurse) [inline docs/server-versions-dependencies.md:26]:
```suggestion
    - Maintenance support, for major versions, will be provided for at least 18 months after a GA release, and there will be at least 6 months' notice before EOL/deprecating support.
```

File: docs/server-versions-dependencies.md [hunk 12]
```diff
@@ -10,6 +10,23 @@ This page details some of the version specific requirements and dependencies nee
...
+
+This versioning policy is intended to serve as a minimum promise to the many customers who rely on us for their critical business operations. If you have any further questions on specifics, please do not hesitate to reach out via community or customer channels.
```
Comment (flossypurse) [inline docs/server-versions-dependencies.md:28]:
```suggestion
- This versioning policy is intended to serve as a minimum set of guidelines for customers who rely on us for their critical business operations.
- If you have any further questions, please do not hesitate to reach out via community or customer channels.
```
Maybe move these up to the top of the versioning policy section and add them with the other general policy bullet points.

**Review by flossypurse — CHANGES_REQUESTED** [review]
Excellent content.

I think we should avoid using "-" in the middle of sentences.

Make sure there are periods at the end of sentences, even bullet points. 

Use sentence casing for titles & headers.

File: docs/server-versions-dependencies.md (hunk 5 above)
Comment (sw-yx) [reply docs/server-versions-dependencies.md:21]:
```suggestion
```

File: docs/server-versions-dependencies.md (hunk 7 above)
Comment (sw-yx) [reply docs/server-versions-dependencies.md:23]:
```suggestion
```

File: docs/server-versions-dependencies.md [hunk 13]
```diff
@@ -10,6 +10,23 @@ This page details some of the version specific requirements and dependencies nee
...
+    - We aim to publish incremental upgrade guides for each minor and major version, which will include specifics around dependency upgrades that we have tested for (e.g. Cassandra 3.0 -> 3.11).
+    - We will offer maintenance support of the last 3 **minor** versions after a release, and do not plan to "backport" patches beyond that.
+    - We will offer maintenance support of **major** versions for at least 12 months after a GA release, and provide at least 6 months' notice before EOL/deprecating support.
+**Temporal Cloud**
+    - All customers will automatically be upgrade to the latest minor version.
+    - Maintenance support, for major versions, will be provided for at least 18 months after a GA release, and there will be at least 6 months' notice before EOL/deprecating support.
+
+- This versioning policy is intended to serve as a minimum set of guidelines for customers who rely on us for their critical business operations.
+- If you have any further questions, please do not hesitate to reach out via community or customer channels.
+
```
Comment (sw-yx) [inline docs/server-versions-dependencies.md:29]:
```suggestion
- All Temporal Server releases abide by the [Semantic Versioning Specification](https://semver.org/).
- Fairly precise upgrade paths and support have been established starting from Temporal `v1.7.0`.
- We provide maintenance support for previously published minor and major versions by continuing to release critical bug fixes related to security, the prevention of data loss, and reliability, whenever they are found.

**Temporal Server OSS**

    - We aim to publish incremental upgrade guides for each minor and major version, which will include specifics around dependency upgrades that we have tested for (e.g. Cassandra 3.0 -> 3.11).
    - We will offer maintenance support of the last 3 **minor** versions after a release, and do not plan to "backport" patches beyond that.
    - We will offer maintenance support of **major** versions for at least 12 months after a GA release, and provide at least 6 months' notice before EOL/deprecating support.

**Temporal Cloud**
    - All customers will automatically be upgrade to the latest minor version.
    - Maintenance support, for major versions, will be provided for at least 18 months after a GA release, and there will be at least 6 months' notice before EOL/deprecating support.

This versioning policy is intended to serve as a minimum set of guidelines for customers who rely on us for their critical business operations. If you have any further questions, please do not hesitate to reach out via community or customer channels.

```

**Review by flossypurse — APPROVED** [review]
Awesome -

**Pull request merged**